TestFoundation/TestNSCharacterSet.swift

```diff
@@ -17,7 +15,93 @@ import SwiftFoundation
 import SwiftXCTest
 #endif
 
+private struct Box {
+    fileprivate let ns: NSCharacterSet
+    fileprivate let swift: CharacterSet
+    
+    private init(ns: NSCharacterSet, swift: CharacterSet) {
+        self.ns = ns
+        self.swift = swift
+    }
+    
+    init(charactersIn string: String) {
+        self.ns = NSCharacterSet(charactersIn: string)
+        self.swift = CharacterSet(charactersIn: string)
+    }
+    
+    static var alphanumerics: Box {
+        return Box(ns: NSCharacterSet.alphanumerics._bridgeToObjectiveC(),
+                   swift: CharacterSet.alphanumerics)
+    }
+    
+    static var decimalDigits: Box {
+        return Box(ns: NSCharacterSet.decimalDigits._bridgeToObjectiveC(),
+                   swift: CharacterSet.decimalDigits)
+    }
+}
+
+private func assertEqual(_ lhs: Box,
+                         _ rhs: Box,
+                         _ message: @autoclosure () -> String = "",
+                         file: StaticString = #file,
+                         line: UInt = #line) {
+    
+    assert(equal: true, lhs, rhs, message, file: file, line: line)
+}
+
+private func assertNotEqual(_ lhs: Box,
+                            _ rhs: Box,
+                            _ message: @autoclosure () -> String = "",
+                            file: StaticString = #file,
+                            line: UInt = #line) {
+    
+    assert(equal: false, lhs, rhs, message, file: file, line: line)
+}
+
+private func assert<T: Equatable>(equal: Bool,
+                                  _ lhs: T,
+                                  _ rhs: T,
+                                  _ message: @autoclosure () -> String = "",
+                                  file: StaticString = #file,
+                                  line: UInt = #line) {
+    
+    if equal {
+        XCTAssertEqual(lhs, rhs, message, file: file, line: line)
+    }
+    else {
+        XCTAssertNotEqual(lhs, rhs, message, file: file, line: line)
+    }
+}
 
+private func assert(equal: Bool,
```

Comment:
If keeping the `Box` is helpful/necessary, then the following comment applies: All of these new asserts seem a bit out of place. Is it not sufficient to make the `Box` `Equatable` and perform these checks in `==`? (The unit tests can then continue using `XCTAssertEqual`.)